classes/Domain/Model/Airport.php

```diff
@@ -23,18 +24,18 @@ class Airport extends Eloquent implements AirportInformationDatabase
     /**
      * @param string $code the IATA Airport Code to get information for
      *
-     * @throws \Exception
+     * @throws EntityNotFoundException
      *
      * @return self
      */
     public function withCode($code): self
     {
         $airport = $this->where('code', $code)->first(['code', 'name', 'country']);
 
-        if (!$airport) {
-            throw new \Exception("An airport matching '{$code}' was not found.");
+        if ($airport instanceof self) {
```

Comment:
What do you think about reverting this change to make it more obvious what the happy path is?

The way it looks right now it suggests that it is an exception if we have actually found an airport, rather than the other way around.